up-l2/dispatchers/README.adoc

```diff
@@ -30,95 +30,88 @@ SPDX-License-Identifier: Apache-2.0
 
 If the uP-L1 delivery method is push:
 
-* *SHALL* provide an API to start/stop dispatching of CEs per-topic, this is to avoid having to queue CEs on the Receiver if the Receiver is not ready to receive the CEs
+* *SHOULD* provide an API to start/stop dispatching of messages per-topic, this is to avoid having to queue messages on the Receiver if the Receiver is not ready to receive
 
 
 == RPC Error Handling
 
-When a dispatcher is unable to dispatch an event for a given reason (queue full, etc...), the dispatcher is responsible to generate an RPC Response message and send it to the originator of the request. The status code is populated in the  link:../messages/v1/README.adoc#_response_message[`commstatus`] attribute of the response message using the <<commstatus-codes>> defined below.
+When a dispatcher is unable to dispatch an event for a given reason (e.g. queue full, etc...), the dispatcher is responsible to generate an RPC Response message and send it to the originator of the request. The status code is populated in the  link:../messages/v1/README.adoc#_response_message[`commstatus`] attribute of the response message using the error codes defined in https://github.com/eclipse-uprotocol/uprotocol-core-api/blob/main/src/main/proto/ustatus.proto[UCode].
 
-.commstatus Codes
-[#commstatus-codes]
-[width="100%",cols="30%,60%",options="header",]
-!===
-|*google.rpc.Code* |*Reason*
 
-|`*UNAVAILABLE*`
-|The req.v1 has expired due to the downstream uE was unavailable (ex. uDevice was disconnected). uE that issued the req.v1 MAY retry with back-off
 
-|`*DEADLINE_EXCEEDED*`
-|CE has timed out per the ttl attribute specifications defined in req.v1 event
+<<rpc-error-flow>> figure below illustrates the sequence of messages for RPC flows and the role dispatchers play in error handling.
 
-|`*PERMISSION_DENIED*`
-|source is not permitted to access sink
+.RPC Error Flow
+[#rpc-error-flow]
+image::rpc_flow.png[RPC Error Handling]
 
-|`*UNAUTHENTICATED*`
-|source does not have valid authentication credentials (ex. uE's identity does not match the source attribute)
 
-|`*RESOURCE_EXHAUSTED*`
-|The dispatcher ran out of resources (buffer full)
+=== uStreamer Throttling of Published Messages
 
-|`*INVALID_ARGUMENT*`
-|Invalid CE header attributes not covered above (ex. any mal-formatted attributes)
+Throttling is the act of reducing the publication frequency of messages published between devices. A subscriber might want to throttle publishations to reduce network bandwidth costs (e.g. cloud app subscribing to vehicle topics like GPS location). 
 
-|`*UNKNOWN*`
-|An unknown (but not critical) error has occurred
+A subscriber uses  `max_frequency` attribute of https://github.com/eclipse-uprotocol/uprotocol-core-api/blob/main/src/main/proto/core/usubscription/v3/usubscription.proto#L121[SubscribeAttributes] message to set the desired maximum publication frequency (value in Hertz), this attribute is only considered for remote topics. 
```

Comment:
Capturing some comments made by Doug

I’d prefer not to specify the throttling in frequency.  Period or a max message rate in seconds/milliseconds would be preferred as this will match our code.  Frequency tends to imply a trend.

Reply:
agree with the above comment. I think units of time is more clear.

> maximum publication frequency

also, I feel using the words "maximum publication" here may lead to confusion. As this is really just a filtering, and the publisher likely won't change its publication frequency